moto/panorama/models.py

```diff
@@ -0,0 +1,304 @@
+        ] = "PANORAMA_APPLIANCE"
+
+    @property
+    def device_aggregated_status(self) -> DEVICE_AGGREGATED_STATUS_TYPE:
+        _device_aggregated_status: DEVICE_AGGREGATED_STATUS_TYPE = self.__device_aggregated_status_manager.status  # type: ignore[assignment]
+        self.__device_aggregated_status_manager.advance()
+        return _device_aggregated_status
+
+    @property
+    def provisioning_status(self) -> PROVISIONING_STATUS_TYPE:
+        _provisioning_status: PROVISIONING_STATUS_TYPE = self.__device_provisioning_status_manager.status  # type: ignore[assignment]
+        self.__device_provisioning_status_manager.advance()
+        return _provisioning_status
+
+    def response_object(self) -> Dict[str, Any]:
+        response_object = super().gen_response_object()
+        response_object = deep_convert_datetime_to_isoformat(response_object)
+        static_response_fields = [
+            "AlternateSoftwares",
+            "Arn",
+            "Brand",
+            "CreatedTime",
+            "CurrentNetworkingStatus",
+            "CurrentSoftware",
+            "Description",
+            "DeviceConnectionStatus",
+            "DeviceId",
+            "LatestAlternateSoftware",
+            "LatestDeviceJob",
+            "LatestSoftware",
+            "LeaseExpirationTime",
+            "Name",
+            "NetworkConfiguration",
+            "SerialNumber",
+            "Tags",
+            "Type",
+        ]
+        return {
+            **{
+                k: v
+                for k, v in response_object.items()
+                if v is not None and k in static_response_fields
+            },
+            **{
+                "DeviceAggregatedStatus": self.device_aggregated_status,
+                "ProvisioningStatus": self.provisioning_status,
+            },
+        }
+
+    def response_listed(self) -> Dict[str, Any]:
+        response_object = super().gen_response_object()
+        response_object = deep_convert_datetime_to_isoformat(response_object)
+        static_response_fields = [
+            "Brand",
+            "CreatedTime",
+            "CurrentSoftware",
+            "Description",
+            "DeviceId",
+            "LastUpdatedTime",
+            "LatestDeviceJob",
+            "LeaseExpirationTime",
+            "Name",
+            "Tags",
+            "Type",
+        ]
+        return {
+            **{
+                k: v
+                for k, v in response_object.items()
+                if v is not None and k in static_response_fields
+            },
+            **{
+                "DeviceAggregatedStatus": self.device_aggregated_status,
+                "ProvisioningStatus": self.provisioning_status,
+            },
+        }
+
+    @property
+    def response_provision(self) -> Dict[str, Union[str, bytes]]:
+        return {
+            "Arn": self.arn,
+            "Certificates": base64.b64encode("certificate".encode("utf-8")).decode(
```

Comment:
We can generate this value once on initialization, and just return that instead - no need to calculate it every time

Reply:
Done: https://github.com/getmoto/moto/pull/6948/commits/194c04a18c5ecb61416ed4f42e9b255165227ae1